src/nplinker/metabolomics/GNPSMolecularFamilyLoader.py

```diff
@@ -0,0 +1,83 @@
+import csv
+from typing import Tuple, List
+from nplinker.logconfig import LogConfig
+from nplinker.metabolomics.IMolecularFamilyLoader import IMolecularFamilyLoader
+from nplinker.metabolomics.molecular_family import MolecularFamily
+from nplinker.metabolomics.singleton_family import SingletonFamily
+
+
+logger = LogConfig.getLogger(__file__)
+
+
+class GNPSMolecularFamilyLoader(IMolecularFamilyLoader):
+    def __init__(self, filename: str):
+        self._families: List[MolecularFamily] = []
+        
+        for family_id, spectra_ids in _load_molecular_families(filename).items():
+            if family_id == -1:
+                for spectrum_id in spectra_ids:
+                    family = SingletonFamily()
+                    family.spectra_ids = [spectrum_id]
+                    self._families.append(family)
+            else:
+                family = MolecularFamily(family_id)
+                family.spectra_ids = spectra_ids
+                self._families.append(family)
+    
+    def families(self) -> List[MolecularFamily]:
+        return self._families
+
+
+def _load_molecular_families(filename: str) -> dict:
```

Comment:
better to annotate returned value as `dict[int, set[str]]`

Reply:
Annotated it correctly as `dict[int, set[int]]`